### DISCUSSION TITLE
`wp_head()` Equivalent in Mecha

### DISCUSSION THREAD

**Discussion by taufik-nurrohman** [mecha-cms/mecha#115] [Ideas]
Mecha makes use of PHP output buffer by default, so you can manipulate the buffer result right before it is shown as the response body.

We have `content` hook to change the buffer result as you like:

~~~ php
Hook::set('content', function($content) {
    $append = '<!-- Append this comment! -->';
    return strtr($content, ['</head>' => $append . '</head>']);
});
~~~

Example code to insert something right before  the `</body>` tag:

~~~ php
Hook::set('content', function($content) {
    $append = 'Put Google Analytic script here!';
    return strtr($content, ['</body>' => $append . '</body>']);
});
~~~

---

If your intention is to insert asset files such as CSS and JavaScript code, you can utilize the `Asset` class. These feature implementations may be more similar to the `wp_enqueue_script()` function in WordPress:

~~~ php
Asset::set('.\path\to\file.css');
Asset::set('.\path\to\file.js');
~~~

~~~ php
Asset::script('alert(\'Yo!\');');
Asset::style(':root { color: #f00; }');
~~~

**Comment by ghost** [comment]
Kalau kita ingin menyematkan, misalnya dua JavaScript eksternal yang berbeda bagaimana penerapannya, Mas?

**Reply by taufik-nurrohman** [reply]
Tinggal ditumpuk saja:

~~~ php
​Asset​::​set​(​'.\path\to\file-1.js'​);
​Asset​::​set​(​'.\path\to\file-2.js'​);
​Asset​::​set​(​'.\path\to\file-3.js'​);
~~~

Kalau urutan pada skrip penting, prioritas bisa ditentukan pada parameter ke dua:

~~~ php
​Asset​::​set​(​'.\path\to\file-1.js', 1​);
​Asset​::​set​(​'.\path\to\file-2.js', 2​);
​Asset​::​set​(​'.\path\to\file-3.js'​, 3);
~~~
